api/notd/collection_processor.py

```diff
@@ -26,8 +29,18 @@ def __init__(self, requester: Requester, ethClient: EthClientInterface, openseaA
         self.erc721MetdataContractAbi = erc721MetdataContractJson['abi']
         self.erc721MetadataNameFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'name'][0]
         self.erc721MetadataSymbolFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'symbol'][0]
+        with open('./contracts/IERC1155MetadataURI.json') as contractJsonFile:
+            erc1155MetadataContractJson = json.load(contractJsonFile)
+        self.erc1155MetadataContractAbi = erc1155MetadataContractJson['abi']
+        self.erc1155MetadataUriFunctionAbi = [internalAbi for internalAbi in self.erc1155MetadataContractAbi if internalAbi.get('name') == 'uri'][0]
+        with open('./contracts/IERC165.json') as contractJsonFile:
+            erc165MetadataContractJson = json.load(contractJsonFile)
+        self.erc165MetadataContractAbi = erc165MetadataContractJson['abi']
+        self.erc165SupportInterfaceUriFunctionAbi = [internalAbi for internalAbi in self.erc165MetadataContractAbi if internalAbi.get('name') == 'supportsInterface'][0]
 
     async def retrieve_collection(self, address: str) -> RetrievedCollection:
+        doesSupportErc721 = (await self.ethClient.call_function(toAddress=address, contractAbi=self.erc165MetadataContractAbi, functionAbi=self.erc165SupportInterfaceUriFunctionAbi, arguments={'interfaceId': _INTERFACE_ID_ERC721}))[0]
```

Comment:
i think this can throw exceptions (like in the token processor). try it with cryptopunks for example I'm sure it will give an error